tests/Phinx/Db/Adapter/SqlServerAdapterTest.php

```diff
@@ -50,6 +50,14 @@ public function testConnection()
         $this->assertSame(\PDO::ERRMODE_EXCEPTION, $this->adapter->getConnection()->getAttribute(\PDO::ATTR_ERRMODE));
     }
 
+    public function testConnectionWithDsnOptions()
+    {
+        $options = $this->adapter->getOptions();
+        $options['dsn_options'] = ['TrustServerCertificate' => 'true'];
+        $this->adapter->setOptions($options);
+        $this->assertInstanceOf('PDO', $this->adapter->getConnection());
+    }
```

Comment:
This test isn't overly valuable as it doesn't really test that the `dsn_options` were used in any way, but there's currently not a good way to do that. This method is also inline with how we test the effect of [port] missing as well:

https://github.com/cakephp/phinx/blob/48d8e1f686852882a319e6deff1ae6748d04ab5c/tests/Phinx/Db/Adapter/SqlServerAdapterTest.php#L62-L68

so it's not like this is the only place we have this. I'd like to address this in a separate PR, so I'm fine with merging this for now as is.